phac-nml/irida-next
# Fix for memberships on group transfer

## What does this PR do and why?
Fixes the memberships after a group transfer, so that the access level is ascending through its descendants.
Project transfers and job creation is out of scope in this PR.
Fixes part of #226.

## Screenshots or screen recordings
![Screenshot from 2023-10-19 14-04-19](https://github.com/phac-nml/irida-next/assets/325703/28fb0520-90d9-47aa-aa65-011bc9b1499c)
![Screenshot from 2023-10-19 14-05-15](https://github.com/phac-nml/irida-next/assets/325703/ae91aa26-5ad6-47a7-8b9a-3eb2a087769a)
![Screenshot from 2023-10-19 14-06-25](https://github.com/phac-nml/irida-next/assets/325703/d617448c-492e-4382-abef-0241332fdf1e)

## How to set up and validate locally
1. Create a new group called `Group A`.
2. Add user `user1@email.com` with `Maintainer` access to `Group A`.
3. Create a new group called `Group B`.
4. Add user `user1@email.com` with `Guest` access to `Group B`.
5. Transfer `Group B` to parent group `Group A`.
6. Verify that `user1@email.com` has changed to `Maintainer` in `Group B`.

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
